### PR TITLE
Bug 1389292 - Add test for bookmarks buffer/local root view

### PR DESCRIPTION
**Pull request by garvankeeley**
Added test case for adding subfolder to remote mobile folder
and navigating it.

### CODE REVIEW

File: UITests/BookmarksPanelTests.swift [hunk 1]
```diff
@@ -83,26 +103,45 @@ class BookmarksPanelTests: KIFTestCase {
...
-        EarlGrey.select(elementWithMatcher: grey_accessibilityLabel("Bookmarks")).inRoot(grey_kindOfClass(NSClassFromString("UITableView")!)).perform(grey_tap())
+        // TEST: Add a local bookmark, and then navigate back to the root view (the navigation will refresh the table).
+        let isAdded = (bookmarks as! MergedSQLiteBookmarks).local.addToMobileBookmarks(URL(string: "http://another-unused")!, title: "123", favicon: nil)
+        XCTAssert(isAdded.value.isSuccess)
+        navigateBackInTableView()
         // is this in the root?
-        EarlGrey.select(elementWithMatcher: grey_accessibilityLabel("123")).assert(grey_notNil())
+        assertRowExists(withTitle: "123")
+        
+        // TEST: Add sub-folder to MobileFolderGUID, ensure it is navigable
```
Comment (garvankeeley) [inline UITests/BookmarksPanelTests.swift]: I should have split up the PR, this part is the only new code, the rest is just refactoring this test code for reuse.

File: UITests/BookmarksPanelTests.swift [hunk 2]
```diff
@@ -61,20 +61,40 @@ class BookmarksPanelTests: KIFTestCase {
...
 
+    private func navigateBackInTableView() {
+        EarlGrey.select(elementWithMatcher: grey_accessibilityLabel("Bookmarks")).inRoot(grey_kindOfClass(NSClassFromString("UITableView")!)).perform(grey_tap())
+    }
+    
+    private func navigateFolder(withTitle title: String) {
+        EarlGrey.select(elementWithMatcher: grey_accessibilityLabel(title)).perform(grey_tap())
+    }
+    
+    private func makeBookmark(guid: String, parentID: String, title: String) -> BookmarkMirrorItem {
```
Comment (rnewman) [inline UITests/BookmarksPanelTests.swift]: We have a little type alias hanging around: `GUID`. Use that instead of `String` where appropriate, like here.

**Comment by garvankeeley** [comment]
@st3fan I can't land this because the 'squash and merge' doesn't enable, I am below some level of permissions I think. Can I have the required access to click that button without status checks completing?

**Comment by rnewman** [comment]
It's greyed out because the build hasn't completed. It might have stalled out.

Push again with that `GUID` change, and see if the build completes 😁

**Comment by garvankeeley** [comment]
Oops, just got reviewed! Sorry, I was browsing the PR list and assumed this was old, then saw the button disabled, and well, I should just eat my dinner...

File: UITests/BookmarksPanelTests.swift (hunk 2 above)
Comment (garvankeeley) [reply UITests/BookmarksPanelTests.swift]: thanks will do!

Comment (garvankeeley) [reply UITests/BookmarksPanelTests.swift]: Functions like this are changed to the much clearer: 
`func makeFolder(guid: GUID, parentID: GUID, title: String, children: [GUID]) -> BookmarkMirrorItem {`

**Pull request merged**